backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/business_rule/event/BusinessRuleEventProcessorCommand.java

```diff
@@ -227,6 +253,79 @@ private static Integer getRecordIdFromSql(final String sql, final int sourceReco
 		return targetRecordId;
 	}
 
+	private static TargetRecordInfo createTargetRecordInfo(@NonNull final AdTableId targetTableId, @NonNull final Integer targetRecordId, @NonNull final String targetTableName, @NonNull final String keyColumnName)
+	{
+		final TargetRecordInfo.TargetRecordInfoBuilder targetRecordInfoBuilder = TargetRecordInfo.builder()
+				.targetRecordRef(TableRecordReference.of(targetTableId, targetRecordId));
+
+		final POInfo targetPOInfo = POInfo.getPOInfo(targetTableName);
+
+		if(targetPOInfo == null)
+		{
+			return targetRecordInfoBuilder
+					.build();
+		}
+
+		if (targetPOInfo.hasColumnName(InterfaceWrapperHelper.COLUMNNAME_DocumentNo))
+		{
+			final String documentNoSQL = "SELECT " + InterfaceWrapperHelper.COLUMNNAME_DocumentNo + " FROM " + targetTableName + " WHERE " + keyColumnName + "=?";
+
+			final String documentNo = DB.getSQLValueStringEx(ITrx.TRXNAME_None, documentNoSQL, targetRecordId);
+			if (documentNo != null)
+			{
+				targetRecordInfoBuilder.documentNo(documentNo);
+			}
+		}
+
+		if (targetPOInfo.hasColumnName(InterfaceWrapperHelper.COLUMNNAME_Value))
+		{
+			final String valueSQL = "SELECT " + InterfaceWrapperHelper.COLUMNNAME_Value + " FROM " + targetTableName + " WHERE " + keyColumnName + "=?";
+
+			final String value = DB.getSQLValueStringEx(ITrx.TRXNAME_None, valueSQL, targetRecordId);
+			if (value != null)
+			{
+				targetRecordInfoBuilder.value(value);
+			}
+		}
+
+		if (targetPOInfo.hasColumnName(InterfaceWrapperHelper.COLUMNNAME_Name))
+		{
+			final String nameSQL = "SELECT " + InterfaceWrapperHelper.COLUMNNAME_Name + " FROM " + targetTableName + " WHERE " + keyColumnName + "=?";
+
+			final String name = DB.getSQLValueStringEx(ITrx.TRXNAME_None, nameSQL, targetRecordId);
+			if (name != null)
+			{
+				targetRecordInfoBuilder.name(name);
+			}
+		}
+		return targetRecordInfoBuilder
+				.build();
+	}
+
+	private @NonNull String getAvailableRecordData(@NonNull final TargetRecordInfo targetRecordInfo)
```

Comment:
imho this method should actually be part o the TargetRecordInfo class.

Reply:
method no longer needed since we use documentSummary